rayon-core/src/sleep/mod.rs

```diff
@@ -20,14 +37,51 @@ const ROUNDS_UNTIL_SLEEPY: usize = 32;
 const ROUNDS_UNTIL_ASLEEP: usize = 64;
 
 impl Sleep {
-    pub fn new() -> Sleep {
+    pub fn new(worker_count: usize) -> Sleep {
         Sleep {
             state: AtomicUsize::new(AWAKE),
-            data: Mutex::new(()),
+            data: Mutex::new(SleepData {
+                worker_count,
+                active_threads: worker_count,
+                blocked_threads: 0,
+                external_waiters: 0,
+            }),
             tickle: Condvar::new(),
         }
     }
 
+    #[inline]
+    pub fn add_external_waiter(&self) {
```

Comment:
I'd like to see a comment on what an "external waiter" is

Reply:
Added.